app/lib/features/attachments/widgets/attachment_item.dart

```diff
@@ -95,18 +123,19 @@ class AttachmentItem extends ConsumerWidget {
     final msgContent = attachment.msgContent();
     final fileName = msgContent.body();
     final fileNameSplit = fileName.split('.');
-    final title = fileNameSplit.first;
+    final title = attachment.name().toString();
     final fileExtension = fileNameSplit.last;
     String fileSize = getHumanReadableFileSize(msgContent.size() ?? 0);
     return Column(
       crossAxisAlignment: CrossAxisAlignment.start,
       children: [
         if (attachmentType == AttachmentType.link) ...[
-          Text(
-            attachment.name() ?? '',
-            maxLines: 1,
-            overflow: TextOverflow.ellipsis,
-          ),
+          if (attachment.name() != null && attachment.name()!.isNotEmpty)
+            Text(
+              attachment.name()!,
+              maxLines: 1,
+              overflow: TextOverflow.ellipsis,
+            ),
```

Comment:
and then this should be
```suggestion
          if (title != null && title.isNotEmpty)
            Text(
              title,
              maxLines: 1,
              overflow: TextOverflow.ellipsis,
            ),
```